### ISSUE TITLE
NaNs seen with transformers version >= 4.21.0 when running HF BERT fine-tuning with XLA_USE_BF16=1

### ISSUE THREAD

**Issue by jeffhataws** [aws-neuron/aws-neuron-sdk#593]
When running HuggingFace BERT (any size) [fine-tuning tutorial](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuronx/tutorials/training/finetune_hftrainer.html) with transformers version >= 4.21.0 and using XLA_USE_BF16=1, I see NaNs in the loss immediately at the first step. The issue is seen also with XLA_DOWNCAST_BF16=1.

The workaround is to use 4.20.0 or earlier. The tutorials currently recommend version 4.15.0.

The issue is also seen on GPU XLA. The NaNs likely come from the transformers library change: https://github.com/huggingface/transformers/pull/17306 . More detail on the issue is at https://github.com/pytorch/xla/issues/4152 .

To reproduce, run the [fine-tuning tutorial](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuronx/tutorials/training/finetune_hftrainer.html) but using transformers version 4.21.0 or newer.

**Comment by jeffhataws** [comment]
This issue was resolved with HuggingFace BERT version >=4.25.1. A work-around was also provided in the [HF fine-tuning tutorial](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuronx/tutorials/training/finetune_hftrainer.html) for  HuggingFace BERT version < 4.25.1 :

```
# Workaround for NaNs seen with transformers version >= 4.21.0
# https://github.com/aws-neuron/aws-neuron-sdk/issues/593
if os.environ.get("XLA_USE_BF16") or os.environ.get("XLA_DOWNCAST_BF16"):
    transformers.modeling_utils.get_parameter_dtype = lambda x: torch.bfloat16
```

**Comment by ShailavTaneja** [comment]
We had the same issue, the NaNs are due to overflow - and the real issue is: 
https://pytorch.org/xla/release/r2.6/perf/amp.html#amp-for-xla-gpu-best-practices
Do not set XLA_USE_F16 flag when using AMP on Cuda devices. This will override the per-operator precision settings provided by AMP and cause all operators to execute in float16.

Stop using` XLA_DOWNCAST_BF16=1 XLA_USE_BF16=1` altogether, 
instead use `model.to(torch.bfloat16)`

I'm trying to get the documentation updated to highlight this. 
https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/appnotes/torch-neuronx/migration-from-xla-downcast-bf16.html
